### PR TITLE
Allow utf-8 characters in comments

### LINKED ISSUE #16553: [Feature Request][compiler] Support UTF8 in comments
Currently, the Move parser doesn't support UTF8 in the source. This is for security reasons and [common in smart contract languages](https://chatgpt.com/share/681adf27-c9dc-8005-b67f-13478d286ee7) for security reasons. However, it is ok to have it in comments, which would be beneficial for example for the Chinese community.

### PR DESCRIPTION
**Pull request by junxzm1990**
## Description

Enable compiler v2 to allow utf-8 characters in comments of move code. 

The compiler’s V2 parser already supports UTF-8 characters within comments. This pull request refines the character validation process by relocating the global character checker: instead of applying it across the entire source code file, it now targets only the code tokens.

Close #16553 

## How Has This Been Tested?
* Existing compiler test cases and transactional test cases
* New [test cases](https://github.com/aptos-labs/aptos-core/tree/jun/support-utf8-comment/third_party/move/move-compiler-v2/tests/checking/utf8-comment) 

## Key Areas to Review
* If any edge cases are missed

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [x] Move Compiler
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-compiler-v2/tests/more-v1/parser/invalid_character_comment.exp [hunk 1]
```diff
@@ -1,7 +1,2 @@
 
-Diagnostics:
-error: invalid character
-  ┌─ tests/more-v1/parser/invalid_character_comment.move:3:44
-  │
-3 │ // Non-ASCII characters in comments (e.g., ф) are also not allowed.
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/tests/more-v1/parser/invalid_character_comment.exp:6]: We should change this comment in the test file (not allowed -> allowed).

File: third_party/move/move-compiler-v2/legacy-move-compiler/src/parser/mod.rs [hunk 2]
```diff
@@ -180,15 +180,7 @@ fn parse_file(
...
-    let buffer = match verify_string(file_hash, &source_buffer) {
-        Err(ds) => {
-            diags.extend(ds);
-            files.insert(file_hash, (fname, source_buffer));
-            return Ok((vec![], MatchedFileCommentMap::new(), diags, file_hash));
-        },
-        Ok(()) => &source_buffer,
-    };
-    let (defs, comments) = match parse_file_string(compilation_env, file_hash, buffer) {
+    let (defs, comments) = match parse_file_string(compilation_env, file_hash, &source_buffer) {
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/legacy-move-compiler/src/parser/mod.rs:183]: We should also test the doc generator `third_party/move/move-prover/move-docgen/src/docgen.rs` by adding some tests where comments have non-ASCII characters.

File: third_party/move/move-compiler-v2/tests/checking/utf8-comment/utf-8-bad-constant-string.exp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+Diagnostics:
+error: invalid byte string
+  ┌─ tests/checking/utf8-comment/utf-8-bad-constant-string.move:4:30
+  │
+4 │        const C: vector<u8> = "abc 这个是代码注释";
```
[the remark this answers is not in the document]
Comment (junxzm1990) [reply third_party/move/move-compiler-v2/tests/checking/utf8-comment/utf-8-bad-constant-string.exp:6]: I intentionally made this one to make sure I did not interrupt the original behavior. The one you revised has been included as `utf-8-in-constant-string.move` -:)

File: third_party/move/move-compiler-v2/legacy-move-compiler/src/parser/comments.rs [hunk 4]
```diff
@@ -14,6 +14,7 @@ pub type FileCommentMap = BTreeMap<(u32, u32), String>;
 
 // We restrict strings to only ascii visual characters (0x20 <= c <= 0x7E) or a permitted newline
 // character--\r--,--\n--or a tab--\t.
+#[allow(unused)]
```
[the remark this answers is not in the document]
Comment (junxzm1990) [reply third_party/move/move-compiler-v2/legacy-move-compiler/src/parser/comments.rs]: I removed the function, but the module redefines several types that are reused across different files. Hence, I kept the file as a placeholder for those type redefines.
